pgrx-macros/src/lib.rs

```diff
@@ -624,18 +624,14 @@ fn example_return() -> pg_sys::Oid {
 
 */
 #[proc_macro_attribute]
+#[track_caller]
 pub fn pg_extern(attr: TokenStream, item: TokenStream) -> TokenStream {
     fn wrapped(attr: TokenStream, item: TokenStream) -> Result<TokenStream, syn::Error> {
-        let pg_extern_item = PgExtern::new(attr.clone().into(), item.clone().into())?;
+        let pg_extern_item = PgExtern::new(attr.into(), item.into())?;
         Ok(pg_extern_item.to_token_stream().into())
     }
 
-    wrapped(attr, item).unwrap_or_else(|e| {
-        let msg = e.to_string();
-        TokenStream::from(quote! {
-          compile_error!(#msg);
-        })
-    })
+    wrapped(attr, item).unwrap_or_else(|e: syn::Error| e.to_compile_error().into())
```

Comment:
I think you were unconsciously programming against syn 1.0 here: this is borrowing, but there's an `into_compile_error()` we could be using here.

Reply:
Ah, no, I think I copied it from elsewhere in that file. The `.into_compile_error()` is definitely new as of 2.0 though.